File: Ejercicio 19/ej19.py
class Pelicula:
    def __init__(self, titulo="", visto=False):
        self.titulo = titulo
        self.visto = visto

    def was_viwed(self):
        if(self.visto):
            print(f"Ya viste {self.titulo}")
        else:
            print(f"Aún no viste {self.titulo}")

title1 = "Venom"
title2 = "Cyberpunk"
title3 = "asda"
title4 = "Top Gun"

p1 = Pelicula("Venom", True)
p2 = Pelicula("Merlina", True)
p3 = Pelicula("Cyberpunk", False)
p4 = Pelicula("WoW", False)
p5 = Pelicula("Top Gun", True)

films = [p1, p2, p3, p4, p5]

def mostrar(films : list, tit : str):
    titulos = [x.titulo for x in films]
    if tit in titulos:
        busqueda(films,tit)
    else:
        print(f"{tit} no se encuentra en la lista")

def busqueda(films, tit):
    for f in films:
        if(f.titulo == tit):
            f.was_viwed()
            
mostrar(films, title1)

File: Ejercicio 19/ej19.js
class Pelicula {
    constructor(titulo, visto) {
      this.titulo = titulo;
      this.visto = visto;
    }

    wasViwed() {
        if(this.visto){
            console.log(`ya viste ${this.titulo}`);
        }else{
            console.log(`aún no viste ${this.titulo}`);
        }
    }
}

p1 = new Pelicula("Venom", true);
p2 = new Pelicula("Merlina", false);
p3 = new Pelicula("asds", true);

lista = [p1, p2, p3];

function busqueda(lista, titulo){
    for (const peli of lista) {
        if(titulo == peli.titulo){
            peli.wasViwed();
        }       
    }
}

function mostrar(lista, titulo){
    titulos = []
    for (const i of lista) {
        titulos.push(i.titulo);
    }

    if(titulos.includes(titulo)){
        busqueda(lista, titulo);
    }else{
        console.log("no se encuentra esa pelicula en lista");
    }
}

mostrar(lista, "Merlina");